python/cppyy/numba_ext.py

```diff
@@ -130,14 +160,57 @@ def get_call_type(self, context, args, kwds):
         except KeyError:
             pass
 
-        ol = CppFunctionNumbaType(self._func.__overload__(*(numba2cpp(x) for x in args)), self._is_method)
+        args_cpp = []
+        for i, arg in enumerate(list(args)):
+            # If the user explicitly passes an argument using numba CPointer, the regex match is used
+            # to detect the pass by reference since the dispatcher always returns typeref[val*]
+            match = re.search(r"typeref\[(.*?)\*\]", str(arg))
```

Comment:
I wonder whether this can be made more "numba-style" by creating new types (e.g. by derivation) instead of string processing.